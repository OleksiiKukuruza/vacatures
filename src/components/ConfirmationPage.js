import React from 'react';
import PropTypes from 'prop-types';
import OrderValues from './OrderValues';
import FeatureList from './FeatureList';

const ConfirmationPage = ({
  vacancy: {
    name,
    originalPrice,
    discount,
    features
  }
}) => (
  <div>
    {name}
    <FeatureList features={features} />
    <OrderValues
      name={name}
      originalPrice={originalPrice}
      discount={discount}
    />
  </div>
);

ConfirmationPage.propTypes = {
  vacancy: PropTypes.shape({
    name: PropTypes.string.isRequired,
    originalPrice: PropTypes.number.isRequired,
    discount: PropTypes.number.isRequired,
    features: PropTypes.arrayOf(PropTypes.string).isRequired
  }).isRequired
};

export default ConfirmationPage;
