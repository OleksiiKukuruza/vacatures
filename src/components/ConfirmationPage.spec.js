import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationPage  from './ConfirmationPage';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('ConfirmationPage', () => {
  it('renders without crashing', () => {
    const props = {
      vacancy: {
        id: '1',
        name: 'name',
        originalPrice: 100,
        discount: 10,
        image: 'img.src',
        features: ['First', 'Second']
      }
    };
    const { wrapper } = setup(ConfirmationPage, props);
    expect(wrapper).toMatchSnapshot();
  });
});
