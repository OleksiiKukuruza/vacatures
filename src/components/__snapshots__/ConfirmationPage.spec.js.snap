// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPage renders without crashing 1`] = `
<div>
  name
  <FeatureList
    features={
      Array [
        "First",
        "Second",
      ]
    }
  />
  <OrderValues
    discount={10}
    name="name"
    originalPrice={100}
  />
</div>
`;
